src/net_processing.cpp

```diff
@@ -394,7 +425,7 @@ class PeerManagerImpl final : public PeerManager
         EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
 
     /** Send a version message to a peer */
-    void PushNodeVersion(CNode& pnode);
+    void PushNodeVersion(CNode& pnode, Peer& peer);
```

Comment:
nit:
```suggestion
    void PushNodeVersion(CNode& pnode, const Peer& peer);
```
Or like cory suggested this could also just be a bool.